superset/templates/superset/models/database/macros.html

```diff
@@ -40,7 +40,7 @@
           name: $('#database_name').val(),
           impersonate_user: $('#impersonate_user').is(':checked'),
           extras: JSON.parse($("#extra").val()),
-          encrypted_extra: JSON.parse($("#encrypted_extra").val()),
+          encrypted_extra: $("#encrypted_extra").val() ? JSON.parse($("#encrypted_extra").val()) : {},
```

Comment:
can we  `const encryptedExtra = $("#encrypted_extra").val();` above the try block first?

Reply:
@mistercrunch - good point. I just pushed that change.